toolkit/tools/pkg/imagecustomizerlib/extractpartitions.go

```diff
@@ -190,3 +220,48 @@ func generateRandom128BitNumber() ([SkippableFramePayloadSize]byte, error) {
 	}
 	return randomBytes, nil
 }
+
+// Construct outputPartitionMetadata object.
+func constructOutputPartitionMetadata(diskPartition diskutils.PartitionInfo, partitionNum int, partitionFilepath string, output *[]outputPartitionMetadata) (err error) {
```

Comment:
This function should return `(outputPartitionMetadata, error)` and the caller should handle the append.